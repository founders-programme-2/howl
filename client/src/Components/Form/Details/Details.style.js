import styled from 'styled-components';

const ErrMsg = styled.p`
  color: red;
  font-weight: bold;
`;

const BoldText = styled.span`
  font-weight: bold;
`;

export { ErrMsg, BoldText };
