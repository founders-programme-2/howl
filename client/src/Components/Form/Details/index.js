import React, { Fragment } from 'react';
import { months, locations } from '../../data.json';
import ErrMsg from './Details.style';

import {
  withStyles,
  Radio,
  FormControlLabel,
  FormControl,
  FormLabel,
  FormGroup,
  RadioGroup,
  InputLabel,
  Select,
  MenuItem,
} from '../../muIndex';
import styles from './muiStyles';

const Details = ({
  classes,
  radio,
  radioChange,
  location,
  dropdownChange,
  locationErr,
  yearErr,
  year,
  month,
}) => {
  const locationQuestion = locationErr ? (
    <ErrMsg>{locationErr}</ErrMsg>
  ) : (
    <p>Where did this event occur?</p>
  );

  const DateQuestion = yearErr ? (
    <ErrMsg>{yearErr}</ErrMsg>
  ) : (
    <p>
      Approximately when did this event occur? If the event spans multiple
      years, select the year when it started.
    </p>
  );

  const yearsGenerator = () => {
    const years = [];
    for (let i = 1960; i <= 1990; i += 1) {
      years.push(i);
    }
    return years;
  };

  const renderSelectItems = items => {
    return items.sort().map(item => (
      <MenuItem key={item} value={item}>
        {item}
      </MenuItem>
    ));
  };

  return (
    <Fragment>
      <h2>Your Contribution</h2>
      <p>
        Collecting personal stories, pictures and artifacts from real women is
        hugely important to us at HOWL. We value everyone&#39;s input and thank
        you in advance for sharing with us. We will now go through and ask you a
        series of questions about the story or image that you&#39;d like to
        contribute. Don&#39;t worry if you can&#39;t remember something
        perfectly. Answer to the best of your ability and put any comments in
        &#39;Additional comments&#39; section at the end of the form.
      </p>
      {locationQuestion}
      <FormControl className={classes.formControl}>
        <InputLabel htmlFor="location" className={classes.label}>
          Location (required)
        </InputLabel>
        <Select
          inputProps={{
            name: 'location',
            id: 'location',
          }}
          aria-label="location"
          value={location}
          onChange={dropdownChange}
          disableUnderline
          className={classes.selectDropdown}
        >
          {renderSelectItems(locations)}
        </Select>
      </FormControl>
      {DateQuestion}
      <FormControl className={classes.formControlDate}>
        <InputLabel htmlFor="month" className={classes.label}>
          Month (optional)
        </InputLabel>
        <Select
          inputProps={{
            name: 'month',
            id: 'month',
          }}
          aria-label="month"
          value={month}
          onChange={dropdownChange}
          disableUnderline
          className={classes.selectDropdown}
        >
          {months.map(monthEle => {
            return (
              <MenuItem key={monthEle} value={monthEle}>
                {monthEle}
              </MenuItem>
            );
          })}
        </Select>
      </FormControl>
      <FormControl
        className={`${classes.formControlDate} ${classes.formControlYear}`}
      >
        <InputLabel htmlFor="year" className={classes.label}>
          Year (required)
        </InputLabel>
        <Select
          inputProps={{
            name: 'year',
            id: 'year',
          }}
          aria-label="year"
          value={year}
          onChange={dropdownChange}
          disableUnderline
          className={classes.selectDropdown}
        >
          {renderSelectItems(yearsGenerator())}
        </Select>
      </FormControl>
      <p>What kind of contribution would you like to make?</p>
      <FormControl className={classes.formControlRadio}>
        <FormLabel component="legend">I would like to share:</FormLabel>

        <RadioGroup
          name="storyType"
          className={classes.group}
          value={radio}
          onChange={radioChange}
        >
          <FormGroup row>
            <FormControlLabel
              value="textPost"
              control={<Radio />}
              label="Text Post"
            />
            <FormControlLabel
              value="imagePost"
              control={<Radio />}
              label="Image Post"
            />
            <FormControlLabel value="both" control={<Radio />} label="Both" />
          </FormGroup>
        </RadioGroup>
      </FormControl>
    </Fragment>
  );
};

export default withStyles(styles)(Details);
