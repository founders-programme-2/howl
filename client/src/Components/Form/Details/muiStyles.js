// TODO: Refactor media query to take in constant variable for size

const styles = () => ({
  selectDropdown: {
    width: '100%',
    paddingLeft: '2%',
  },
  formControl: {
    marginLeft: '5%',
    marginRight: '5%',
    width: '50%',
    background: 'white',
    boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.25)',
    borderRadius: 4,
    '@media (max-width: 768px)': {
      width: '80%',
    },
  },
  formControlDate: {
    background: 'white',
    display: 'inline-block',
    marginLeft: '5%',
    marginRight: '5%',
    marginBottom: '2%',
    width: '50%',
    boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.25)',
    borderRadius: 4,
    '@media (max-width: 768px)': {
      width: '80%',
    },
  },

  formControlMonth: {
    marginBottom: 0,
  },

  label: {
    paddingLeft: '2%',
  },
  formControlRadio: {
    marginLeft: '5%',
    marginRight: '5%',
    width: '50%',
  },
});

export default styles;
