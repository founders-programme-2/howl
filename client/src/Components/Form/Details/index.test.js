import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Details from './index';

test('Natural', () => {
  let state = {};
  const props = {
    radio: '',
    radioChange: event => {
      state = { ...state, radio: event.target.value };
    },
    location: '',
    selectedDate: new Date(),
    textChange: name => event => {
      state = { ...state, [name]: event.target.value };
    },
    handleDateChange: date => {
      const options = {
        year: 'numeric',
        month: 'long',
      };
      const dateModified = date.toLocaleDateString('en-US', options);
      const splitted = dateModified.split(' ');
      state = { ...state,
        selectedDate: dateModified,
        month: splitted[0],
        year: splitted[1],
      }
    },
    dropdownChange: event => {
      state = { ...state, [event.target.name]: event.target.value };
    }
  }
  const details = renderer.create(<BrowserRouter> <Details { ...props } /> </BrowserRouter>).toJSON();
  expect(details).toMatchSnapshot();
});
