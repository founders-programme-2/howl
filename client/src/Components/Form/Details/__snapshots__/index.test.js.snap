// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Natural 1`] = `
Array [
  " ",
  <h2>
    Your Contribution
  </h2>,
  <p>
    Collecting personal stories, pictures and artifacts from real women is hugely important to us at HOWL. We value everyone's input and thank you in advance for sharing with us. We will now go through and ask you a series of questions about the story or image that you'd like to contribute. Don't worry if you can't remember something perfectly. Answer to the best of your ability and put any comments in 'Additional comments' section at the end of the form.
  </p>,
  <p>
    Where did this event occur?
  </p>,
  <div
    className="MuiFormControl-root Details-formControl-2"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Details-label-3"
      data-shrink={false}
      htmlFor="location"
    >
      Location (required)
    </label>
    <div
      aria-label="location"
      className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl Details-selectLocation-1"
      onClick={[Function]}
    >
      <div
        className="MuiSelect-root"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="select-location"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          id="location"
          name="location"
          type="hidden"
          value=""
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>,
  <p>
    Approximately when did this event occur?
  </p>,
  <div
    className="MuiGrid-root DateFun-grid-76"
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
        data-shrink={true}
      >
        Year and Month
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
        onClick={[Function]}
      >
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-194 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-195"
            style={
              Object {
                "width": 0,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value="May 2019"
        />
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <p
        className="MuiFormHelperText-root DateFun-helperText-75 MuiFormHelperText-contained MuiFormHelperText-filled"
      >
        When did this happen?
      </p>
    </div>
  </div>,
  <p>
    What kind of contribution would you like to make?
  </p>,
  <div
    className="MuiFormControl-root Details-formControlRadio-4"
  >
    <legend
      className="MuiFormLabel-root"
    >
      I would like to share:
    </legend>
    <div
      className="MuiFormGroup-root"
      role="radiogroup"
    >
      <div
        className="MuiFormGroup-root MuiFormGroup-row"
      >
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-253 MuiRadio-root MuiRadio-colorSecondary"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <div
                className="PrivateRadioButtonIcon-root-257"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-258"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
              <input
                checked={false}
                className="PrivateSwitchBase-input-256"
                disabled={false}
                name="storyType"
                onChange={[Function]}
                type="radio"
                value="textPost"
              />
            </span>
          </span>
          <span
            className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
          >
            Text Post
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-253 MuiRadio-root MuiRadio-colorSecondary"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <div
                className="PrivateRadioButtonIcon-root-257"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-258"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
              <input
                checked={false}
                className="PrivateSwitchBase-input-256"
                disabled={false}
                name="storyType"
                onChange={[Function]}
                type="radio"
                value="imagePost"
              />
            </span>
          </span>
          <span
            className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
          >
            Image Post
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-253 MuiRadio-root MuiRadio-colorSecondary"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <div
                className="PrivateRadioButtonIcon-root-257"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-258"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
              <input
                checked={false}
                className="PrivateSwitchBase-input-256"
                disabled={false}
                name="storyType"
                onChange={[Function]}
                type="radio"
                value="both"
              />
            </span>
          </span>
          <span
            className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
          >
            Both
          </span>
        </label>
      </div>
    </div>
  </div>,
  " ",
]
`;
